### PR TITLE
Implement namespace.follow for symbol types

### PR DESCRIPTION
**Pull request by xbauch**
In `driver.adb` call `Follow_Irep` on every symbol in symbol table.
In `ireps_generator.py` reuse `To_JSON` to implement `Follow_Irep`: traverse the irep tree to follow `I_Symbol_Type` nodes. The irep table is not modified: we only change what ireps are referred to in some node (of the irep tree).

### CODE REVIEW

File: gnat2goto/driver/driver.adb [hunk 1]
```diff
@@ -51,6 +52,8 @@ package body Driver is
    procedure Translate_Standard_Types;
    procedure Initialize_CProver_Internal_Variables (Start_Body : Irep);
    procedure Add_CProver_Internal_Symbols;
+   procedure Follow_Type_Declarations (Old_Table : Symbol_Table;
+                                       New_Table : in out Symbol_Table);
```
[the remark this answers is not in the document]
Comment (xbauch) [reply gnat2goto/driver/driver.adb]: Well I'm rewriting the symbols and I don't think I can modify a hash map while iterating over it. The second part: yes, will rewrite to just `out`.

File: gnat2goto/driver/driver.adb [hunk 2]
```diff
@@ -379,4 +383,34 @@ package body Driver is
...
+      function Follow_Symbol (I : Irep) return Irep;
+      function Follow_Symbol (I : Irep) return Irep is
+      begin
+         return Follow_Symbol_Type (I, Old_Table);
+      end Follow_Symbol;
+   begin
+      for Sym_Iter in Old_Table.Iterate loop
+         declare
+            Ignored : Boolean;
+            Unused  : Symbol_Maps.Cursor;
```
[the remark this answers is not in the document]
Comment (xbauch) [reply gnat2goto/driver/driver.adb]: I copied it from a function above this one, but ok, this sounds more descriptive.

File: gnat2goto/driver/tree_walk.adb [hunk 3]
```diff
@@ -1835,6 +1835,8 @@ package body Tree_Walk is
          when E_Record_Subtype => Do_Itype_Record_Subtype (N),
          when E_Signed_Integer_Type => Do_Itype_Integer_Type (N),
          when E_Floating_Point_Type => Create_Dummy_Irep,
+         when E_Anonymous_Access_Type => Make_Pointer_Type
```
[the remark this answers is not in the document]
Comment (xbauch) [reply gnat2goto/driver/tree_walk.adb]: Yes, although it is required to make the `primitive_pointer` test work. So, separate commit or separate PR?

File: gnat2goto/ireps/ireps_generator.py [hunk 4]
```diff
@@ -937,9 +937,9 @@ def generate_code(self, optimize, schema_file_names):
         # Emit spec and body file
         s = new_file("ireps.ads")
         b = new_file("ireps.adb")
-        write(s, "with Types;         use Types;")  # Source_Ptr
+        write(s, "with Types;             use Types;")  # Source_Ptr
         write(s, "")
-        write(s, "with GNATCOLL.JSON; use GNATCOLL.JSON;")  # JSON
+        write(s, "with GNATCOLL.JSON;     use GNATCOLL.JSON;")  # JSON
```
[the remark this answers is not in the document]
Comment (xbauch) [reply gnat2goto/ireps/ireps_generator.py]: Shouldn't be here, will remove.

File: gnat2goto/ireps/ireps_generator.py [hunk 5]
```diff
@@ -1596,6 +1612,15 @@ def generate_code(self, optimize, schema_file_names):
         continuation(b)
         write(b, "")
 
+        write(b, "function Follow_Irep (L : Irep_List;")
+#        with indent(b):
```
[the remark this answers is not in the document]
Comment (xbauch) [reply gnat2goto/ireps/ireps_generator.py]: Sorry, will be removed.

File: gnat2goto/ireps/ireps_generator.py [hunk 6]
```diff
@@ -1816,6 +1841,98 @@ def generate_code(self, optimize, schema_file_names):
         write(b, "end To_JSON;")
         write(b, "")
 
+        write_comment_block(b, "Follow_Irep")
+        write(b, "function Follow_Irep (I : Irep;")
```
[the remark this answers is not in the document]
Comment (xbauch) [reply gnat2goto/ireps/ireps_generator.py]: You mean another python script?

File: gnat2goto/driver/tree_walk.adb (hunk 3 above)
[the remark this answers is not in the document]
Comment (hannes-steffenhagen-diffblue) [reply gnat2goto/driver/tree_walk.adb]: Preference for me would be separate PR but separate commit at the least

File: gnat2goto/ireps/ireps_generator.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (NlightNFotis) [reply gnat2goto/ireps/ireps_generator.py]: I'm also uncomfortable with how big this new code inserted is. I understand it's codegen stuff, which normally is long, but I'm also under the impression that it has to be factored out into a function, and then called here.

**Comment by NlightNFotis** [comment]
@xbauch The tests show that gnat2goto now crashes on (almost?) all tests? Was it like this before, and/or is it depending on another PR going in?

**Comment by xbauch** [comment]
@NlightNFotis better?

**Comment by hannes-steffenhagen-diffblue** [comment]
Some of the CI failures may be fixed by a rebase (which you'll need anyway)

**Pull request merged**